book/listings/todo_app/2/todo_row/mod.rs

```diff
@@ -0,0 +1,84 @@
+mod imp;
+
+use crate::todo_object::TodoObject;
+use glib::{BindingFlags, Object};
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{glib, pango};
+use pango::{AttrList, Attribute};
+
+glib::wrapper! {
+    pub struct TodoRow(ObjectSubclass<imp::TodoRow>)
+    @extends gtk::Box, gtk::Widget,
+    @implements gtk::Accessible, gtk::Buildable, gtk::ConstraintTarget, gtk::Orientable;
+}
+
+impl Default for TodoRow {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TodoRow {
+    pub fn new() -> Self {
+        Object::new(&[]).expect("Failed to create `TodoRow`.")
+    }
+
+    pub fn bind(&self, todo_object: &TodoObject) {
+        // Get state
+        let imp = imp::TodoRow::from_instance(self);
+        let completed_button = imp.completed_button.get();
+        let content_label = imp.content_label.get();
+
+        // Bind `todo_object.completed` to `todo_row.completed_button.active`
+        let completed_button_binding = todo_object
+            .bind_property("completed", &completed_button, "active")
+            .flags(BindingFlags::SYNC_CREATE | BindingFlags::BIDIRECTIONAL)
+            .build()
+            .expect("Could not bind properties");
+        // Save binding
+        imp.bindings.borrow_mut().push(completed_button_binding);
```

Comment:
Is it possible to borrow_mut this once at the top and just use that?

Reply:
Yes, that works!